Polish focus and active styles, and do cleanups

jasmussen opened the pull request:
This PR does three things:

1. It fixes a regression with the :active style for buttons. Now it works again:

![active](https://user-images.githubusercontent.com/1204802/43578204-5c97e1c4-964e-11e8-9176-5ba4998efc23.gif)

2. It replaces in a whole SLEW of files, the use of 1px for border widths, to use an SCSS variable. This is a cleanup job suggested by @tofumatt, and shouldn't affect anything. I don't expect we'll ever actually want to change the value of the $border-width, but by using a SCSS variable, we can make complex SCSS math easier to understand. For example https://github.com/WordPress/gutenberg/commit/832a6c1a83247e2a812b56e749d0c88dceb794ed#diff-b40a015680110faaabe748b80f5b4cc9L32 no longer needs an HTML comment.

3. It redesigns input fields. This is what needs design feedback. Before they were square and ligth gray, now they are rounded and darker, which fixes #7053. I also changed the focus style to be bolder — this may feel heavier at first hand, but with the darker base input style I felt like this was the only way to give contrast between base and focus style. This also fixes any issues with showing input fields on gray backgrounds, as noted in https://github.com/WordPress/gutenberg/issues/7053#issuecomment-405095702. 

![new focus styles](https://user-images.githubusercontent.com/1204802/43578381-ea5a3840-964e-11e8-916e-ed5de3ab3534.gif)

![embed](https://user-images.githubusercontent.com/1204802/43578593-7f01852a-964f-11e8-98a6-6dc96bb2a593.gif)

At first glance these new styles might feel a bit too heavy handed, but after a little while the reduced amount of color in the focus style actually comes across as simpler. Give it a little while to work on  you.

Commit:
Fix regression with button active style.

Commit:
Move from 1px to using $border-width

This is sort of some cleanup work across the files.

Commit:
Try new input focus style.

Fixes #7053.

afercia commented:
3, the darker border for input fields is very welcomed from an accessibility perspective, pending design feedback. In #7053 / #8037 a different solution is explored (darkening only the bottom border), which honestly I personally don't like that much.

With regards to this PR, I have only one consideration so far: the comment 
`// Lightest gray that can be used for AAA contrast.`

is a bit misleading for two reasons: it's AA level and 3:1 is the minimum contrast for **non-text**. So it should be changed to:
`// Lightest gray that can be used for AA non-text contrast.`

Worth reminding that for **text** contrast the minimum contrast ratio is 4.5:1 (with exceptions for large text) so the lightest gray that can be used is `#6c7781`.

chrisvanpatten commented:
From a design perspective my only question is whether the border-radius should match for buttons and inputs, but overall the new design looks great. The darker border is actually kinda nice! 👍

jasmussen commented:
Good catch Andrea, I'll revise the comment and add the other note as well. 

> From a design perspective my only question is whether the border-radius should match for buttons and inputs

This is a good catch. I'll noodle on this — the radius is the same as it is for the new buttons, but not for the old buttons. I'll see if I can unify all three somehow, or at least reduce to two.

karmatosed reviewed (approved):
I really like this and think it would be a benefit as is to get in. We can iterate from there. Thanks for working on this.

tofumatt commented on edit-post/assets/stylesheets/_mixins.scss:
```diff
@@ -166,15 +166,22 @@
 // Tabs, Inputs, Square buttons
 @mixin input-style__neutral() {
-	outline-offset: -1px;
+	//outline-offset: -1px;
```
🔥

tofumatt commented on edit-post/assets/stylesheets/_mixins.scss:
```diff
@@ -166,15 +166,22 @@
 	color: $dark-gray-900;
-	outline: 1px solid $blue-medium-600;
-	box-shadow: 0 0 0 2px $blue-medium-200;
+	//outline: 1px solid $blue-medium-600;
```
🔥

tofumatt commented on edit-post/assets/stylesheets/_mixins.scss:
```diff
@@ -166,15 +166,22 @@
+	border-color: $blue-medium-500;
+	box-shadow: 0 0 0 1px $blue-medium-500;
+
+	// Windows High Contrast mode will show this outline, but not the box-shadow
```
Comments should end with a period.

tofumatt commented on edit-post/assets/stylesheets/_variables.scss:
```diff
@@ -57,5 +57,5 @@ $block-parent-side-ui-clearance: $parent-block-padding - $block-padding; // spac
 // Buttons & UI Widgets
-$button-style__radius-roundrect: 4px;
-$button-style__radius-round: 50%;
+$radius-roundrect: 4px;
```
I know it was like that when you got here, but since we're changing it anyway, could we make this `$radius-round-rectangle`? `roundrect` is quite unfriendly.

tofumatt commented on packages/editor/src/components/post-permalink/style.scss:
```diff
@@ -9,16 +9,16 @@
 	// put toolbar snugly to edge on mobile
-	margin-left: -$block-padding - 1px;	// stack borders
-	margin-right: -$block-padding - 1px;
+	margin-left: -$block-padding - $border-width;	// stack borders
```
That's a weird indentation and doesn't really quite explain this, but I think it's saying "This stacks the border on top of another border"?

I dunno, leave it if it's not clear, but since this line is already changing it'd be nice to tidy it up. 🤷‍♂️

tofumatt reviewed (approved):
Looks great. There are a few little tweaks worth making before merge. Just a few commented-out lines that should be removed and a few cleanup requests if you can make them.

After that, please do 🚢

jasmussen commented:
Thanks for the reviews. Addressed feedback in https://github.com/WordPress/gutenberg/pull/8385/commits/2f61bb558a5f474aba3168ce9e58856c609f60cb. Rebased and hoping the tests pass.

Noting though, that this input field change also affects checkboxes:

![category](https://user-images.githubusercontent.com/1204802/43627875-72828ebc-96f8-11e8-82d3-7420b67f7f9b.gif)

But I don't think that's a problem, quite the opposite actually.

tofumatt commented:
Last night tests were failing a bunch (as were local installs using `bin/install-wordpress.sh`) because of a weird Docker permission thing. If CI is still failing on `Installing WordPress...` that's the same error... I guess we'll have to fix it up, but I have no idea what caused it... 😢

jasmussen commented:
Also through in a micro fix for a double border:

![screen shot 2018-08-03 at 08 46 50](https://user-images.githubusercontent.com/1204802/43628253-d2eb7e70-96f9-11e8-9691-f8d20451da35.png)

Now:

![screen shot 2018-08-03 at 08 46 37](https://user-images.githubusercontent.com/1204802/43628255-d65eea38-96f9-11e8-963b-5f6b97f2a161.png)

jasmussen commented:
![screen shot 2018-08-03 at 08 50 19](https://user-images.githubusercontent.com/1204802/43628358-46b4cdac-96fa-11e8-816d-efdc4c093a15.png)

Tests pass locally, though I'm on an old version of the dev environment where I can't run the e2e tests locally (I know, I'll get to it). Would appreciate if someone can verify whether Travis is making mistakes or whether a legit test is failing.

tofumatt commented:
It's legit, sort of: #8418.

I'll see why the Docker container is messed up, but I'm not really sure why that's happening.

jasmussen commented:
Merging as tests pass and this is approved, yay!

ktmn commented:
Any particular reason for the 4px border radius? Everything else is 0px except buttons, I kind of preferred it that way.
For example when a select is surrounded by text the lack of border radius sort of makes it look more like a singular unit (although that's very subjective).
![](https://i.imgur.com/FXlAEzs.png)
Side note: the dropdown arrow doesn't have the best spacing either and seems distorted (browser specific?).

I mean, maybe there's a case for it here:
![](https://i.imgur.com/KwW1L5a.png)
so the inputs look similar to buttons, but they don't look worse without border radius either.


Also `input[type=text]` is has 31px height while `select` has 28px
![](https://i.imgur.com/tbIbh8E.png)

jasmussen commented:
@ktmn I agree the select box could use some love.

The answer is that yes, there's a reason. The reason is that it's so far the best input field design we've found that:

- Has sufficient contrast against a white background
- Looks like an input field, even when showing in otherwise square gray-bordered boxes

Gutenberg introduces a number of new design elements, and notably a number of accessibility improvements compared to WordPress core. We've tried a few other designs for input boxes that met these requirements, but so far the rounded version was the only one that passed the test for meeting all the requirements and notably _feeling right_ in Gutenberg.

We could try reducing the border radius to 3px, perhaps. This is the same radius that stock buttons use.

karmatosed commented:
A border radius even is small does have a softening effect I think is a positive impact visually. I am ok experimenting reducing, however I think overall it's a positive step.
